feat: Add About Page

vgeorge opened the pull request:
Contributes to https://github.com/NASA-IMPACT/veda-ui/issues/1562.

**This is work in progress**

Changes added:

- Setup Next.js to render mdx files
- Copied [about page mdx content](https://github.com/NASA-IMPACT/veda-config-eic/blob/c7d6399153f7e31c33bff6f03490cb70be1b27f0/overrides/about.mdx) from current template

Know issues:

- The page uses `<Block>` and `<Prose>` components, using the wrapper in @lib file to make them client components. But on the server log an errors `ReferenceError: window is not defined` is displayed
- Browser log display several `Theme Value Error: path...` errors, which might be related to some missing provider

dzole0311 commented on app/about/page.mdx:
```diff
@@ -0,0 +1,124 @@
+
+<Block>
+	<Prose>
```
@vgeorge I think we can ditch the `Block` and `Prose` components from veda-ui completely and just directly use p tags, heading tags etc with the appropriate USWDS util classes. 

Like that, if we modify the tokens (e.g. font colors, sizes, etc) in the uswds-theme.scss file, the changes will be applied to all the USWDS classes used throughout.

dzole0311 replied:
Not sure if the page also needs to be an .mdx file (these are mostly static) 🤷 I'd personally opt in towards just keeping the content under an about .tsx route.

vgeorge replied:
Ok, I'll keep as is.

vgeorge commented:
Status update:

- Added AboutLayout to wrap styles for the about page
- Added SideNav component

Next steps:

- Make SideNav sticky as it should display the current active section (see https://github.com/NASA-IMPACT/veda-ui/issues/1561)
- Make SideNav react to scroll
- Improve content styles

vgeorge commented:
Changes added:

- Sidenav is now sticky. It currently stops at top=0, but we can adjust once the header is merged
- Sidenav observes the sections and make the active the one closer to the top
- Added custom styles for mdx content

Know issues:

- No mobile styles, I suggest we add in separate
- The last sections of the mdx are short and wouldn't not appear as selected if the viewport is also short

Preview:

![about](https://github.com/user-attachments/assets/e3e0fdca-593b-4ef3-8078-d7bea1de0f6c)

Suggested review steps:

- Run the app, visit home page
- Click on the about link at main nav
- About page content should load
- Scrolling updates the active section in the SideNav
- Sidenav is sticky, always visible
- Clicking on Sidenav updates the URL and the visible section
- Visiting a [section URL](http://localhost:3000/about#datasets) directly activates the proper sidenav section

This is ready for review.

AliceR commented on app/about/side-nav.tsx:
```diff
@@ -0,0 +1,85 @@
+
+    sectionIds.forEach((id) => {
+      const element = document.getElementById(id);
+      if (element) observer.observe(element);
```
Should we add an `else` to warn about the missing element? For example 
```
console.warn(`Element with ID '${id}' not found in the DOM);
```
Might be better then just silently skipping, for debugging.

AliceR replied:
Ups, you apparently did not see this comment. Maybe not that important though, now that we generate the list of ids from actual page content.

AliceR commented on app/about/side-nav.tsx:
```diff
@@ -0,0 +1,85 @@
+        }
+      },
+      {
+        rootMargin: '-10% 0px -70% 0px',
```
Did you intend to use `--page-start-offset`? Not sure if those actually relate though...?

vgeorge replied:
removed

AliceR commented on app/about/side-nav.tsx:
```diff
@@ -0,0 +1,85 @@
+                : ''
+            }
+          >
+            {id.charAt(0).toUpperCase() + id.slice(1).replace(/-/g, ' ')}
```
This line could benefit from a helper function that clearly names what should happen here, with a comment or even unit test to document input -> expected output.

vgeorge replied:
I added a utility class `text-capitalize` to make it more readable.

AliceR commented on app/about/page.mdx:
```diff
@@ -0,0 +1,84 @@
+<section id="resources">
+  <h3>U.S. Government Resources on Earth's Changes and Impacts:</h3>
+
+  - **EPA**: https://www.epa.gov/data/environmental-dataset-gateway
```
Those urls are not styled as a url, not clickable and not blue/underlined as they should be per design.

AliceR commented on app/about/page.mdx:
```diff
@@ -0,0 +1,84 @@
+</section>
+
+<section id="next">
+  <h3>Next Steps</h3>
```
The design I am looking at on [Figma](https://www.figma.com/design/2Vd0SDhFOTBJaYioJPr0Ei/Earth.gov?node-id=2432-25751&t=cXUWCT3vbNtFlT2F-0) says: "What's coming next?", maybe one of us has an outdated reference?

vgeorge replied:
fixed

AliceR reviewed (changes requested):
This is looking good @vgeorge ! Really cool how the sidenav updates on scroll! I just have a couple of comments, no major concern though. 

🚀

vgeorge commented on app/about/layout.tsx:
```diff
@@ -0,0 +1,26 @@
+  children: React.ReactNode;
+}) {
+  return (
+    <div className='grid-container'>
```
I didn't noticed the library included a <Grid> component. I suggest we go ahead with the current implementation and switch if we notice a benefit from using the component.

vgeorge commented on app/about/side-nav.tsx:
```diff
@@ -0,0 +1,85 @@
+      <p className='margin-y-2'>On this page</p>
+      <SideNav
+        items={sectionIds.map((id, i) => (
+          <a
```
I don't think so, they are inpage URLs which doesn't seem to benefit from Next.js Link optimizations.

vgeorge commented on app/about/page.mdx:
```diff
@@ -0,0 +1,84 @@
+<section id="why">
+  <h3>Why is the global understanding of Earth and its interconnected systems important?</h3>
```
The ids are needed because  the section headers are long and cannot be abbreviated to the nav labels in the designs.

vgeorge commented:
@AliceR thanks, I added changes addressing your comments and this is ready for another review.
